app/views/users/show.html.erb

```diff
@@ -1 +1,30 @@
-<h1>Single User data will be shown here</h1>
\ No newline at end of file
+<div>
+  <div>
+    <%= image_tag @user.photo alt: "User's pic here" %>
+    <div>
+      <h1><%= @user.name %></h1>
+      <p>Total posts: <%= @user.posts_counter %></p>
+    </div>
+  </div>
+  <div>
+    <span>Bio</span>
+    <p><%= @user.bio %></p>
+  </div>
+
+  <% if @user.three_most_recent_posts.empty? %>
+    <div>This user has not post.</div>
+  <% else %>
+    <div>
+      <% @user.three_most_recent_posts.each_with_index do |post, index| %>
+        <div>
+          <div><%= link_to "Post  ##{index + 1}", user_post_path(post.author, post) %></div>
+          <div><%= post.text %></div>
+          <div>Comments: <%= post.comments_counter %>, Likes: <%= post.likes_counter %></div>
+        </div>
+      <% end %>
+    </div>
+  <% end %>
```

Comment:
- [ ] It seems bugs are prevailing in the codebase as I encountered another one. Since the link wasn't working I tried to manually change the URL to redirect to the user show page but the below exceptions are thrown and that's because you're trying to access an undefined method `three_most_recent_posts` please refer back to the user model as it's defined as `recent_3_posts` instead.
- Here's the error shown:
  ![image](https://github.com/Aqurds/coding-thought/assets/104968236/37d872f3-a21a-4d1e-a524-f97f5c39cff9)
### Note for future reviewers 📝  
- [ ] Since I wasn't able to open the user show page please make sure the layout matches the one shown in the [wireframe here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_user_page.png) and the page functions properly.

Reply:
Fixed